day_10/task_2/tests/pipe_maze_tests.cpp

```diff
@@ -129,6 +129,45 @@ std::vector<MazeScopeParams> GenerateMazeScopeParams()
     };
 }
 
+INSTANTIATE_TEST_SUITE_P(IsInMazeScope,
+                         MazeScopeTests,
+                         testing::ValuesIn(GenerateMazeScopeParams()));
+
+
+TEST_P(MazeScopeTests, GivenMazePoint_WhenColorNeighborForPoint_ExpectColorEightPoint)
+{
+    // const PipeMaze pipe_maze = {
+    // "- L | F 7",
+    // "7 S - 7 |",
+    // "L | 7 | |",
+    //  "- L - J |",
+    //  "L | -J F"};
+
+    auto [color_right, color_top_right, color_down_right, color_top, color_down,
+          color_left, color_top_left, color_down_left] = GetParam();
+
+    maze.color_neighbor(MazePoint{1, 1});
```

Comment:
I think, it is important to **colour neighbors depends on the origin point and the direction** , so we would need to pass also second argument which would let implementator determine which direction we go and hence which side is left and which right.

We could go from MazePoint{1,1} into two directions and based on the direction the colouring should change.

Reply:
why? When we have a Maze Point we should return color for all 8 neighbors for it

Reply:
Firstly, we will apply this function only for MazePoints which are part of the loop, hence for every single call of `color_neighbor` we will color at most 6 neighbors, because some two neighbors must be pipes, which are connected to the origin input.

Secondly, imagine when we call the function twice:
maze.color_neighbor(MazePoint{2, 3});
maze.color_neighbor(MazePoint{2, 1});

Then the question for you, how the color_neighbor() should colorize th MazePoint{2,2}?

I would expect to colorize it by the same color, but why? Because if we go from MazePoint{1, 3} to MazePoint{2, 3}, then MazePoint{2, 2} is on the right hand and if we go from MazePoint{3, 1} to MazePoint{2, 1}, then MazePoint{2, 2} is also on the right hand, hence it gets the same color twice. But without passing the second input, which is prev or next maze point, then it is imposible (at least for me) to implement colorizing sides which is accord with right/left sides.

Reply:
ok